src/go/k8s/webhooks/redpanda/validate_redpanda.go

```diff
@@ -0,0 +1,20 @@
+package redpanda
+
+import (
+	"fmt"
+
+	redpandav1alpha1 "github.com/redpanda-data/redpanda/src/go/k8s/apis/redpanda/v1alpha1"
+)
+
+// ValidateRedpanda validates the Redpanda field in spec
+func ValidateRedpanda(console *redpandav1alpha1.Console, cluster *redpandav1alpha1.Cluster) error {
+	if rp := console.Spec.Redpanda; rp != nil {
+		if aa := rp.AdminAPI; aa != nil {
+			// Only mTLS auth is supported for RedpandaAdmin
+			if aa.Enabled && !cluster.IsAdminAPIMutualTLSEnabled() {
+				return fmt.Errorf("redpanda adminApi may be enabled when mTLS is enabled") //nolint:goerr113 // no need to declare new error type
```

Comment:
Maybe it only my english, but the error could say
```suggestion
				return fmt.Errorf("redpanda adminApi in console need to have enabled mTLS") //nolint:goerr113 // no need to declare new error type
```

Reply:
I am following [this](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#validation) convention

Reply:
Then we should use the following:

> When asserting a requirement in the positive, use "must". Examples: "must be greater than 0", "must match regex '[a-z]+'". Words like "should" imply that the assertion is optional, and must be avoided.

```suggestion
				return fmt.Errorf("redpanda adminApi must be set when mTLS is enabled") //nolint:goerr113 // no need to declare new error type
```

Reply:
But redpanda adminApi can be unset even if mTLS is enabled right? It's optional after all, so I think it is not a "must" but a "may"?